extensions/guacamole-auth-json/src/main/java/org/apache/guacamole/auth/json/connection/ConnectionService.java

```diff
@@ -176,6 +177,21 @@ public GuacamoleTunnel connect(UserData.Connection connection,
         String hostname = proxyConfig.getHostname();
         int port = proxyConfig.getPort();
 
+        // handle guacd-[hostname/port/ssl] overrides
+        String overrideGuacdHostname = connection.getGuacdHostname();
+        if (overrideGuacdHostname != null && !overrideGuacdHostname.isEmpty()) {
+            hostname = overrideGuacdHostname;
+        }
+        Integer overrideGuacdPort = connection.getGuacdPort();
+        if (overrideGuacdPort != null) {
+            port = overrideGuacdPort;
+        }
+        EncryptionMethod proxyMethod = proxyConfig.getEncryptionMethod();
+        Boolean overrideGuacdSSL = connection.getGuacdSsl();
```

Comment:
Similarly, I think we prefer `boolean` over `Boolean`, unless specifically requiring `Boolean`.